offer-manager-domain/src/main/groovy/life/qbic/business/offers/identifier/OfferId.groovy

```diff
@@ -1,26 +1,70 @@
 package life.qbic.business.offers.identifier
 
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+
 /**
- * Describes the offer identifier
+ * Represents and identifier for an offer
  *
- * Contains the random part, the project conserved part and the version from which the offer identifier is build
+ * Contains the random part, project conserved part and the version of an identifier
  *
  * @since 0.1.0
  */
-class OfferId extends TomatoId{
+@CompileStatic
+@EqualsAndHashCode(includeFields = true)
+class OfferId implements Comparable<OfferId>{
+
+    private static final String ALPHABET = "abcdefghijklmnopqrstuvwxyz"
+
+    private String projectPart
+    private String randomPart
+    private int version
+
+    OfferId(String projectPart, int version) {
+        this.randomPart = generateFourLetterString()
+        this.projectPart = projectPart
+        this.version = version
+    }
+
+    OfferId(String projectPart, String randomPart, int version) {
+        this.projectPart = projectPart
+        this.randomPart = randomPart
+        this.version = version
+    }
+
+    private static String generateFourLetterString() {
+        StringBuilder randomString = new StringBuilder()
+        Random random = new Random()
+        for (int i = 0; i < 4; i++){
+            def randomIndex = random.nextInt(ALPHABET.size())
+            def newRandomChar = ALPHABET[randomIndex]
+            randomString.append(newRandomChar)
+        }
+        return randomString.toString()
+    }
+
+    void increaseVersion() {
+        this.version += 1
+    }
+
+    String getProjectPart() {
+        return projectPart
+    }
 
-    OfferId(RandomPart randomPart, ProjectPart projectPart, Version versionTag) {
-        super(randomPart, projectPart, versionTag)
+    String getRandomPart() {
+        return randomPart
     }
 
-    OfferId(OfferId offerId) {
-        super(new RandomPart(offerId.randomPart),
-                new ProjectPart(offerId.projectPart),
-                new Version(offerId.version))
+    int getVersion() {
+        return version
     }
 
     @Override
-    String toString() {
-        return "O-${super.projectPart}-${super.randomPart}-${super.version}"
+    int compareTo(OfferId other) {
+        if (this.equals(other)) {
+            // we want to be in sync with the equals method
+            return 0
+        }
+        return this.version <=> other.version
```

Comment:
I dont know this operator, can you enlighten me? :)

Reply:
This is the `compareTo` operator 😄